app/src/main/java/de/westnordost/streetcomplete/data/osm/changes/StringMapChangesBuilder.java

```diff
@@ -27,6 +27,16 @@ public void delete(@NonNull String key)
 		}
 		checkDuplicate(key);
 		changes.put(key, new StringMapEntryDelete(key, valueBefore));
+		deleteIfPresent("source:" + key);
```

Comment:
> I think perhaps this should be in the responsibility of each and every individual quest that deletes things.

I moved it here as deleteIfPresent("source:" + YET_ANOTHER_KEY) after every single deletion looked ugly, but I failed to consider this. I will move it back. It will make this quest code uglier, but I am no longer sure is it overall improving situation.